src/app/portlets/dot-edit-content/dot-edit-content.component.ts

```diff
@@ -18,22 +21,30 @@ export class DotEditContentComponent implements OnInit {
     contentletActionsUrl: SafeResourceUrl;
     dialogTitle: string;
     source: any;
+    pageIdentifier: string;
+
+    private originalValue: any;
+    private isModelUpdate = false;
 
     constructor(
         private dotConfirmationService: DotConfirmationService,
-        private ref: ChangeDetectorRef,
         private route: ActivatedRoute,
         private sanitizer: DomSanitizer,
         private ngZone: NgZone,
         public dotEditContentHtmlService: DotEditContentHtmlService,
+        private dotContainerContentletService: DotContainerContentletService,
         public dotLoadingIndicatorService: DotLoadingIndicatorService,
         private dotMessageService: DotMessageService
     ) {}
 
+
+
     ngOnInit() {
         this.dotLoadingIndicatorService.show();
-        this.route.data.pluck('editPageHTML').subscribe((editPageHTML: string) => {
-            this.dotEditContentHtmlService.initEditMode(editPageHTML, this.iframe);
+
+        this.route.data.pluck('editPageHTML').subscribe((editPageHTML: {render: string, inode: string, identifier: string}) => {
+            this.pageIdentifier = editPageHTML.identifier;
+            this.dotEditContentHtmlService.initEditMode(editPageHTML.render, this.iframe);
 
             this.dotEditContentHtmlService.contentletEvents.subscribe((res) => {
                 switch (res.event) {
```

Comment:
I think it's better that instead of doing the ngZone.run in each method, wrap the switch into and have it in one place, that should work.